### PR TITLE
upgraded to scala 2.12 and sbt 0.13.16, bumped versions dependant lib…

### PR DESCRIPTION
**Pull request by toggm**
…raries as well

### CODE REVIEW

**Comment by malcolmgreaves** [comment]
Thanks for this! Just testing locally and will merge soon :D

File: codegen/build.sbt [hunk 1]
```diff
@@ -2,24 +2,24 @@ import com.nitro.build._
 
 import PublishHelpers._
 
+ScriptedPlugin.scriptedSettings
```
Comment (malcolmgreaves) [inline codegen/build.sbt:5]: I don't know about this -- what does this do? (Purely for my own educational desires)

**Comment by malcolmgreaves** [comment]
Hi @toggm ! I found a few errors in this PR. I made some fixes from your branch and pushed up the changes into the branch `mg_updates_for_pr-upgrade_scala_2.12`. [1] Please merge these changes into your branch so that this PR is updated. Once this is done, I will happily squash merge your PR.

Thank you very much for giving this project some TLC :D

[1] https://github.com/malcolmgreaves/avro-codegen/tree/mg_updates_for_pr-upgrade_scala_2.12

File: codegen/build.sbt (hunk 1 above)
Comment (toggm) [reply codegen/build.sbt:5]: I found this one in the following issue description. Had similar issues when I first tried to compile it with another version of sbt:
https://github.com/sbt/sbt/issues/3325

**Comment by toggm** [comment]
Merged your changes into my PR and additionaly fixed some deprecation warnings.
Thanks for reacting that fast. With this I'm looking forward to migrate our own project to Scala 2.12.

**Comment by malcolmgreaves** [comment]
Wonderful! Merging.

**Comment by malcolmgreaves** [comment]
@toggm This is now released on sonatype as:
```
"io.malcolmgreaves" %% "avro-codegen-runtime" % "0.3.6"
```

**Comment by toggm** [comment]
Thank you very much

**Pull request merged**